xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.cpp

```diff
@@ -35,14 +35,54 @@ CRendererDRMPRIME::~CRendererDRMPRIME()
 
 CBaseRenderer* CRendererDRMPRIME::Create(CVideoBuffer* buffer)
 {
-  if (buffer && dynamic_cast<CVideoBufferDRMPRIME*>(buffer) &&
-      CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
-          SETTING_VIDEOPLAYER_USEPRIMERENDERER) == 0)
+  if (buffer && CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+                    SETTING_VIDEOPLAYER_USEPRIMERENDERER) == 0)
   {
-    CWinSystemGbm* winSystem = dynamic_cast<CWinSystemGbm*>(CServiceBroker::GetWinSystem());
-    if (winSystem && winSystem->GetDrm()->GetVideoPlane()->plane &&
-        std::dynamic_pointer_cast<CDRMAtomic>(winSystem->GetDrm()))
-      return new CRendererDRMPRIME();
+    auto buf = dynamic_cast<CVideoBufferDRMPRIME*>(buffer);
+    if (!buf)
+      return nullptr;
+
+    if (!buf->AcquireDescriptor())
+      return nullptr;
+
+    AVDRMFrameDescriptor* desc = buf->GetDescriptor();
+    if (!desc)
+      return nullptr;
+
+    AVDRMLayerDescriptor* layer = &desc->layers[0];
+    uint32_t format = layer->format;
+
+    auto winSystem = static_cast<CWinSystemGbm*>(CServiceBroker::GetWinSystem());
+    if (!winSystem)
+      return nullptr;
+
+    auto drm = std::static_pointer_cast<CDRMAtomic>(winSystem->GetDrm());
+    if (!drm)
+      return nullptr;
+
+    auto modifiers = drm->GetVideoPlaneModifiersForFormat(format);
+    if (modifiers->empty())
+    {
```

Comment:
I'm not 100% sure, but modifiers property is optional. In case of missing modifier property, DRMUtils.cpp sets default modifer (DRM_FORMAT_MOD_LINEAR) only for GUI plane, but not for video plane. So, this will fail on systems which don't expose any modifier.

Reply:
@jernejsk yes you are right. Do you know of any hardware that exposes YUV formats but does not list modifiers?

listing `DRM_FORMAT_MOD_LINEAR` should be safe to add regardless.

Reply:
I have to check to be sure but all Allwinner SoCs with DE2 (and currently DE3) display unit support only linear formats. I think that planes in such case don't expose modifier property.

Reply:
This is modetest output of Allwinner H6: http://ix.io/2iQf Later it will get support for AFBC which will introduce modifiers, but for now, it doesn't have any.

Reply:
Thanks, I've added a fixup.